### PR TITLE
Apply cleanups suggested when bringing this ADO VMSS support to vcpkg

### PR DESCRIPTION
**Pull request by BillyONeal**
* Add documentation to scripts and functions.
* Rename create-new-agent-image.ps1 to create-vmss.ps1, since the output of this is a scale set, not an image.
* Add [CmdletBinding()] to all the things to make PowerShell autocomplete and similar happy.
* Change all Write-Output to Write-Host.
* Remove unnecessary Write-Reminders script output.
* Use a Spot VM for initial provisioning in addition to the scale set.
* Rename Start-WaitForShutdown to Wait-Shutdown to use a PowerShell approved verb.

### CODE REVIEW

**Comment by BillyONeal** [comment]
Most of these changes were suggested by @strega-nil in the vcpkg ingestion of this tech.

File: azure-devops/create-vmss.ps1 [hunk 1]
```diff
@@ -1,10 +1,20 @@
 # Copyright (c) Microsoft Corporation.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 #
```
Comment (StephanTLavavej) [inline azure-devops/create-vmss.ps1:3]: Do we need this empty comment line anymore?

File: azure-devops/create-vmss.ps1 [hunk 2]
```diff
@@ -55,10 +104,26 @@ function New-Password {
...
 }
 
-function Start-WaitForShutdown {
+<#
+.SYNOPSIS
+Waits for the shutdown of the specified resource.
+
+.DESCRIPTION
+Wait-Shutdown takes a VM, and checks if there's a 'PowerState/stopped'
+code; if there is, it returns. If there isn't, it waits ten seconds and
```
Comment (StephanTLavavej) [inline azure-devops/create-vmss.ps1:113]:
```suggestion
code; if there is, it returns. If there isn't, it waits 10 seconds and
```

File: azure-devops/provision-image.ps1 [hunk 3]
```diff
@@ -1,12 +1,42 @@
 # Copyright (c) Microsoft Corporation.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 #
```
Comment (StephanTLavavej) [inline azure-devops/provision-image.ps1:3]: Remove empty comment line?

File: azure-devops/provision-image.ps1 [hunk 4]
```diff
@@ -1,12 +1,42 @@
...
-# Sets up a machine in preparation to become a build machine image, optionally switching to
-# AdminUser first.
+
+<#
+.SYNOPSIS
+Sets up a machine to be an image for a scale set.
+
+.DESCRIPTION
+provision-image.ps1 runs on an existing, freshly provisioned virtual machine,
+and sets that virtual machine up as a vcpkg build machine. After this is done,
```
Comment (StephanTLavavej) [inline azure-devops/provision-image.ps1:11]: First, improve flow. Second, fix apparent copy-paste error?

```suggestion
and sets up that virtual machine as a build machine. After this is done,
```

File: azure-devops/provision-image.ps1 [hunk 5]
```diff
@@ -122,34 +184,60 @@ Function InstallVisualStudio {
...
-    Write-Output 'Failed to install Visual Studio!'
-    Write-Output $_.Exception.Message
-    exit 1
+    Write-Error "Failed to install Visual Studio! $($_.Exception.Message)"
   }
 }
 
+<#
+.SYNOPSIS
+Install a .msi file.
```
Comment (StephanTLavavej) [inline azure-devops/provision-image.ps1:193]:
```suggestion
Install an .msi file.
```

File: azure-devops/provision-image.ps1 [hunk 6]
```diff
@@ -122,34 +184,60 @@ Function InstallVisualStudio {
...
+
+.DESCRIPTION
+InstallMSI takes a url where an .msi lives, and installs that .msi to the system.
```
Comment (StephanTLavavej) [inline azure-devops/provision-image.ps1:196]:
```suggestion
InstallMSI takes a URL where an .msi lives, and installs that .msi to the system.
```

File: azure-devops/provision-image.ps1 [hunk 7]
```diff
@@ -158,90 +246,116 @@ Function InstallZip {
...
   }
 }
 
+<#
+.SYNOPSIS
+Installs NVIDIA's CUDA Toolkit.
+
+.DESCRIPTION
+InstallCuda installs the CUDA Toolkit with the features specified as a
+space separated list of strings in $Features.
```
Comment (StephanTLavavej) [inline azure-devops/provision-image.ps1:324]: Hyphenate compound adjective:

```suggestion
space-separated list of strings in $Features.
```

**Comment by StephanTLavavej** [comment]
Oh yeah - can you summarize what the cleanups are? It doesn't have to be posted in the Changelog or commit message, but our PRs should attempt to explain what they're changing.

File: azure-devops/create-vmss.ps1 [hunk 8]
```diff
@@ -14,10 +24,25 @@ $LiveVMPrefix = 'BUILD'
 $WindowsServerSku = '2019-Datacenter'
 
 $ProgressActivity = 'Creating Scale Set'
-$TotalProgress = 10
+$TotalProgress = 11
```
Comment (CaseyCarter) [inline azure-devops/create-vmss.ps1:27]: https://youtu.be/KOO5S4vxi0o (No change requested.)

File: azure-devops/create-vmss.ps1 [hunk 9]
```diff
@@ -43,6 +81,17 @@ function Find-ResourceGroupName {
   return $result
 }
 
+<#
+.SYNOPSIS
+Creates a randomly generated password.
```
Comment (CaseyCarter) [inline azure-devops/create-vmss.ps1:86]: "Creates" and "generates" are largely redundant here. How about:
```suggestion
Generates a random password.
```

File: azure-devops/create-vmss.ps1 (hunk 1 above)
Comment (BillyONeal) [reply azure-devops/create-vmss.ps1:3]: We never did. I thought you liked it ¯\\\_(ツ)\_/¯

Comment (StephanTLavavej) [reply azure-devops/create-vmss.ps1]: I am not a fan of this anywhere. I do like having an empty line between the usual banner and any normal comments, though.

**Comment by StephanTLavavej** [comment]
When this is merged, https://github.com/microsoft/STL/wiki/Checklist-for-Updating-Toolset-and-or-Dependencies will need to be updated.

**Comment by BillyONeal** [comment]
Thanks Stephan :)

**Comment by StephanTLavavej** [comment]
Thanks for these cleanups and documentation improvements!
